google-cloud-bigquery/acceptance/bigquery/advanced_test.rb

```diff
@@ -0,0 +1,167 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "bigquery_helper"
+
+describe Google::Cloud::Bigquery, :advanced, :bigquery do
+  let(:dataset_id) { "#{prefix}_dataset" }
+  let(:dataset) { bigquery.dataset(dataset_id) || bigquery.create_dataset(dataset_id) }
+  let(:table_id) { "examples_table" }
+  let(:table) { @table }
+
+  before do
+    @table = get_or_create_example_table dataset, table_id
+  end
+
+  it "loads and returns the properly formatted data" do
+    table.data.sort_by { |r| r["id"] }.zip(example_table_rows) do |returned_row, example_row|
+      assert_rows_equal returned_row, example_row
+    end
+  end
+
+  it "queries values in standard mode" do
+    rows = bigquery.query "SELECT * FROM #{dataset_id}.#{table_id} WHERE id = ?", params: [2]
+
+    rows.class.must_equal Google::Cloud::Bigquery::QueryData
+    rows.count.must_equal 1
+    row = rows.first
+
+    assert_rows_equal rows.first, example_table_rows[1]
+  end
+
+  it "queries repeated scalars in legacy mode" do
```

Comment:
There isn't a super clear explanation online. We link to this URL in the docs:

https://cloud.google.com/bigquery/docs/reference/legacy-sql